src/hipscat/pixel_tree/pixel_alignment.py

```diff
@@ -386,3 +388,58 @@ def perform_align_trees(
     if include_all_left and left_index < len(left):
         _add_remaining_pixels(added_until, left, left_index, LEFT_SIDE, mapping)
     return mapping
+
+
+def filter_alignment_by_moc(alignment: PixelAlignment, moc: MOC) -> PixelAlignment:
+    """Filters an alignment by a moc to only include pixels in the aligned_tree that overlap with the moc,
+    and the corresponding rows in the mapping.
+
+    Args:
+        alignment (PixelAlignment): The pixel alignment to filter
+        moc (mocpy.MOC): The moc to filter by
+
+    Returns:
+        PixelAlignment object with the filtered mapping and tree
+    """
+    moc_ranges = moc.to_depth29_ranges
+    tree_29_ranges = alignment.pixel_tree.tree << (2 * (29 - alignment.pixel_tree.tree_order))
+    tree_mask = perform_filter_by_moc(tree_29_ranges, moc_ranges)
+    new_tree = PixelTree(alignment.pixel_tree.tree[tree_mask], alignment.pixel_tree.tree_order)
```

Comment:
```suggestion
    new_tree = filter_by_moc(alignment.pixel_tree)
```

Reply:
I refactored this a bit more to avoid running the moc filter code twice, but let me know if you don't like it

Reply:
I think that all of those four lines could be replaced by the `filter_by_moc` call. Is that not true?

Reply:
We need the `tree_mask` variable for filtering the alignment data frame in the next line.

Reply:
Got it. Then I think the refactoring doesn't add much - can you revert to the original?

Reply:
Sure thing! Done now